searx/searx
# [enh] add DOI resolver from sci-hub / replace default DOI

Read more about DOI at [Wikipedia](https://en.wikipedia.org/wiki/Digital_object_identifier) and at [doi.org](https://www.doi.org/). It's identification system for scientific articles. DOI is to scientific papers what ISBN is to books.
What these DOI resolvers do is that you give them the DOI number of any specific scientific paper(e.g., [10.1109/5.771073](https://doi.org/10.1109/5.771073)) and then they'll give the PDF version of that paper in return; but free of charge, because scientific papers are mostly distributed by a handful of big corporations and are well behind paywalls.


## What does this PR do?

Add DOI resolver domains from sci-hub and replace the default DOI resolver.

## Why is this change important?

The existing default DOI resolver https://sci-hub.tw redirects to insecure side

## How to test this PR locally?
```
make run
```
- Visit config at http://127.0.0.1:8888/config and check default_doi_resolver and doi_resolvers
- Open preferences and activate the "Open Access DOI rewrite" plugin
- Tab "General" select "Open Access DOI resolver"
- Save / run a query in category 'science' and check DOI URLs


## Related issues

Closes #2693, #2260 and #910
